From: Jason Dillaman <dillaman@redhat.com>
Date: Thu, 16 Jun 2016 09:38:28 -0400
Subject: librbd: force-remove journal when disabling feature and removing
 image

If mirroring is enabled, it's nearly certain that the peer rbd-mirror
daemon will be a registered client within the journal. Without
forcing the removal, this would prevent the removal from occurring.

Signed-off-by: Jason Dillaman <dillaman@redhat.com>

Resolves: rhbz#1347195
---
 src/librbd/Journal.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/librbd/Journal.cc b/src/librbd/Journal.cc
index b874005..d4ae06b 100644
--- a/src/librbd/Journal.cc
+++ b/src/librbd/Journal.cc
@@ -424,7 +424,7 @@ int Journal<I>::remove(librados::IoCtx &io_ctx, const std::string &image_id) {
     return r;
   }
 
-  r = journaler.remove(false);
+  r = journaler.remove(true);
   if (r < 0) {
     lderr(cct) << "failed to remove journal: " << cpp_strerror(r) << dendl;
     return r;
